### PR TITLE
VDR 2.3.x

### PR DESCRIPTION
**Pull request by CvH**
This updates VDR to 2.3.x, all plugins can be used like before. It also includes some cleanup.
This is not really runtime tested so consider it as very WIP.

I also added roboTV plugin (an client for Android TV and Nvidia Shield) and DDCI2 plugin that allows to make use of the build in CI of DigitalDevices cards. 
Sadly I can't test anything of that - if someone has the needed HW I am happy if someone could report back (I would drop it again if no feedback appears).

testbuilds (Generic/RPi2/AML)
http://cvh.libreelec.tv/addons/9.0/vdr/

### CODE REVIEW

File: packages/addons/addon-depends/vdr-plugins/vdr-plugin-live/patches/vdr-plugin-live-01-makefile.patch [hunk 1]
```diff
@@ -0,0 +1,35 @@
+diff --git a/Makefile b/Makefile
+--- a/Makefile
++++ b/Makefile
+@@ -11,7 +11,7 @@ PLUGIN := live
+ VERSION := $(shell grep '\#define LIVEVERSION ' setup.h | awk '{ print $$3 }' | sed -e 's/[";]//g')
+ 
+ ### Check for libpcre c++ wrapper
+-HAVE_LIBPCRECPP := $(shell pcre-config --libs-cpp)
++HAVE_LIBPCRECPP := $(shell pcre-config --libs libpcrecpp libpcre)
```
Comment (mglae) [inline packages/addons/addon-depends/vdr-plugins/vdr-plugin-live/patches/vdr-plugin-live-01-makefile.patch]: Typo: `pcre-config` ->  `pkg-config`

File: packages/addons/addon-depends/vdr-plugins/vdr-plugin-live/patches/vdr-plugin-live-01-makefile.patch [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
++CXXFLAGS  += $(shell pkg-config --cflags tntnet)
++LIBS      += $(shell pkg-config --libs tntnet)
+ 
+ ### Optional configuration features
+ PLUGINFEATURES :=
+ ifneq ($(HAVE_LIBPCRECPP),)
+ 	PLUGINFEATURES += -DHAVE_LIBPCRECPP
+-	CXXFLAGS       += $(shell pcre-config --cflags)
++	CXXFLAGS       += $(shell pkg-config --cflags libpcreposix libpcre)
+ 	LIBS           += $(HAVE_LIBPCRECPP)
```
Comment (mglae) [inline packages/addons/addon-depends/vdr-plugins/vdr-plugin-live/patches/vdr-plugin-live-01-makefile.patch]: Enabling live plugin results in:

```Dec 20 22:42:29 LibreELEC vdr[10346]: [10346] ERROR: /storage/.kodi/addons/service.multimedia.vdr-addon/plugin/libvdr-live.so.2.3.8: undefined symbol: _ZN7pcrecpp2RE6no_argE```

Using `LIBS           += -Wl,-Bstatic $(HAVE_LIBPCRECPP) -Wl,-Bdynamic` solves it but hopefully someone knows a less hackish solution.

**Comment by belaid38** [comment]
where is build for wetek play 2

**Comment by CvH** [comment]
AML should work (only LE9 builds) besides what was mentioned by @mglae , didn't had time to rebuild/push fix

**Comment by belaid38** [comment]
arm  or aarch64  

i  tested in arm  with le9 wetek play 2   instaled ok  
not  work  arm  

i need  tested aarch64 le9

**Comment by CvH** [comment]
At LE9 we have only arm, WP2 is running 64bit/32bit - 64bit kernel and 32bit userland.
There is no aarch64 anymore.

**Comment by belaid38** [comment]
@CvH   i  tested with LibreELEC-WeTek_Play_2.arm-9.0-rb-leia10.img build raybuntu  vdr 2.3.x is  work fine good 
scan channel work good
channel 4k work fine
switch port diseqc  not  work  
-------------------------------------------
test LibreELEC-WeTek_Play_2.arm-rb-leia21.img  vdr not work

**Comment by CvH** [comment]
@mglae sorry for this late reply, fixed your comments, updated some version and already tested it a bit. It works for me.

I uploaded updated compiled addons here http://cvh.libreelec.tv/addons/9.0/vdr/

**Comment by InuSasha** [comment]
@CvH when you switch to `PKG_TOOLCHAIN=manual`, we can avoid the empty `makeinstall_target()`

**Comment by CvH** [comment]
@InuSasha pls have a look at my temp commit (also removed some -j1, but this is unrelated)

**Comment by InuSasha** [comment]
the cmake toolchain needs the makeinstall_rarget

**Comment by CvH** [comment]
@InuSasha works without :>

**Comment by InuSasha** [comment]
ok the the install phase is running (i think -> afk)

**Comment by mglae** [comment]
> @mglae sorry for this late reply, fixed your comments, updated some version and already tested it a bit. It works for me.

No need to hurry, we do have a working 2.2.0.  :)

I'm using your update with DVB-C under LE 8.2 for about four weeks. Looks stable. (Did not test the new roboTV and DDCI2 plugins.)

There is only on issue. The vdr needs ~2.7 seconds to shut down on my not slow machine but is terminated after 2 seconds (same for vdr 2.2.0). This may cause problems if it is killed while writing configuration files. Increasing the time out should not harm:

```
diff --git a/packages/addons/service/vdr-addon/source/system.d/service.multimedia.vdr-addon.service b/packages/addons/service/vdr-addon/source/system.d/service.multimedia.vdr-addon.service
index 0c353f1ae..bfbb96b8c 100644
--- a/packages/addons/service/vdr-addon/source/system.d/service.multimedia.vdr-addon.service
+++ b/packages/addons/service/vdr-addon/source/system.d/service.multimedia.vdr-addon.service
@@ -4,7 +4,7 @@ After=graphical.target
 
 [Service]
 ExecStart=/bin/sh -c "exec sh /storage/.kodi/addons/service.multimedia.vdr-addon/bin/vdr.start"
-TimeoutStopSec=2
+TimeoutStopSec=4
 Restart=always
 RestartSec=2
 StartLimitInterval=0
```

Just tested your generic build from stick using streamdev on a Milhouse #0124x, worked.

There is just a message I do not see in LE 8.2 and it's too late for further investigation:
```sh[1221]: vdr.bin: Relink `/usr/lib/libfreetype.so.6' with `/usr/lib/libpthread.so.0' for IFUNC symbol `longjmp'```

**Comment by CvH** [comment]
@mglae added your change - works for me but I am not really using VDR productive :)
- added translation changes (converted to strings.po)
- added service.log written to `/storage/.kodi/userdata/addon_data/service.multimedia.vdr-addon/service.log` makes searching for errors easier (completely forgotten to add this in LE7 days ...)

uploaded changed addon.zip

**Comment by mglae** [comment]
>added service.log written to /storage/.kodi/userdata/addon_data/service.multimedia.vdr-addon/service.log

There are only a few lines written to service.log the remaining are still in the journal. What about adding a $ADDON_HOME/bin/vdrlog with
```
#!/bin/sh
# [common header]
journalctl -b -0 -u service.multimedia.vdr-addon "$@"
```
for easy error checking instead.

>I have the same libfreetype.so.6 error at log, google hints to glibc problems

There is a [glibc 2.26 patch](https://sourceware.org/git/?p=glibc.git;a=commit;h=88758c4ad3f046d050bc2c3ae0f172b6524ca6c2) but I'm not sure if we need a PR?!

**Comment by CvH** [comment]
@mglae hmm not sure how to add it to the cmd line that it actually logs it ? (have a look at commit `temp log`)

btw I saw some restfulapi errors, already fixed one but no idea how to fix the other one (I have no dvb device attached) - this spams the log what makes it basically useless
```
ERROR: invalid primary device number: 1
ERROR: no primary device found - using first device!
ERROR: invalid primary device number: 1
ERROR (ci.c,2990): (null): Bad address
ERROR (tools.c,1595): /usr/share/locale: No such file or directory
```

> There is a glibc 2.26 patch but I'm not sure if we need a PR?!

actually I am pretty scared in adding glibc patches as we can't really predict/test if it create something odd at other packages - the error seems to be cosmetic ( ? ) and should be gone at next bump, from https://sourceware.org/glibc/wiki/Glibc%20Timeline it should be tomorrow

**Comment by mglae** [comment]
@CvH 
>hmm not sure how to add it to the cmd line that it actually logs it ? (have a look at commit temp log)

Systemd logs all syslog() and stderr/stdout automatically.

Please remove the `-p err` to see all messages by default, the advanced user can add the parameter when needed.

>(I have no dvb device attached) - this spams the log what makes it basically useless

VDR exits without any device found and is restarted immediately. You have to activate dummydevice plugin in this case. I've started to modify the start script to solve the issue some time ago and look again on it...

I agree with you, as long there is no crash we better omit the glibc patch.

**Comment by CvH** [comment]
@mglae pushed some changes, I did the logging differently, I added an "Debug" option at the vdr addon settings, if enabled it writes an service.log to the userdata dir (like Tvh is doing it) - see `temp log`, I also updated the uploaded addons

Was also able to disable the `ERROR: no OSD provider available - using dummy OSD!` message spam at log `packages/addons/addon-depends/vdr/patches/vdr-02-disable-logspam.patch`

**Comment by mglae** [comment]
@CvH Just PRed the dummydevice auto select changes to avoid the exit/restart loop when no primary device is found.

It includes changes to the "Debug" option:

- Starting journalctl execution in parallel to exec creates implicit subshells and vdr does not inherit master PID.  Furthermore journalctl has to be started before with an additional delay to see all vdr messages.

- Activate vdr debug messages when selected.

**Comment by CvH** [comment]
@mglae tx for the help, will test a bit and then squash it down to reasonable commits (for final review) :)

**Comment by CvH** [comment]
@mglae cleaned up the commits - looks good and works from my perspective
if you are happy with it I we could merge it soon
- updated http://cvh.libreelec.tv/addons/9.0/vdr/

**Comment by belaid38** [comment]
@CvH   where is he vdr update fix tody for AML ?

**Comment by CvH** [comment]
@belaid38 sorry, due the changes at our buildsystem the aml build need to be completely recompiled from scratch (no time for it yesterday)

**Comment by mglae** [comment]
@CvH  It's working fine (being still at the state of my yesterdays commit).

One final question: needs default-settings.xml to be updated for clean installs?

**Comment by CvH** [comment]
@mglae actually we are also not too sure about that :D
pushed that change (also updated uploads)

**Comment by CvH** [comment]
updated RoboTV,Sat>IP and VNSI plugin to latest
ready to merge whenever someone hits the button :)

**Comment by CvH** [comment]
@chewitt this is not even close to a small or safe change :)

**Pull request merged**